File: src/com/falconraptor/utilities/Random.java
package com.falconraptor.utilities;

import java.time.LocalDateTime;
import java.time.ZoneId;

public class Random {
    public static int[] parseStringArray(String[] s) {
        int[] temp = new int[s.length];
        for (int i = 0; i < s.length; i++) temp[i] = Integer.parseInt(s[i]);
        return temp;
    }

    public static int[] calcTimeLeft(int[] time, int[] date) {
        LocalDateTime order, now = LocalDateTime.now(ZoneId.of("GMT")), temp;
        try {
            order = LocalDateTime.of(date[0], date[1], date[2], time[0], time[1], time[2]);
        } catch (Exception e) {
            e.printStackTrace();
            return new int[]{0, 0, 0};
        }
        double minutes = 0;
        temp = order;
        while (!(temp.getYear() == now.getYear() && temp.getDayOfMonth() == now.getDayOfMonth() && temp.getMonthValue() == now.getMonthValue() && temp.getHour() == now.getHour() && temp.getMinute() == now.getMinute() && temp.getSecond() == now.getSecond())) {
            temp = temp.plusSeconds(1);
            minutes += 1;
        }
        int minute, hour, second;
        hour = (int) minutes / 3600;
        minutes = minutes % 3600;
        minute = (int) minutes / 60;
        second = (int) minutes % 60;
        return new int[]{hour, minute, second};
    }
}

File: src/com/falconraptor/utilities/Random.scala
package com.falconraptor.utilities

import java.time.{LocalDateTime, ZoneId}

object Random {
  def parseStringArray(s: Array[String]): Array[Int] = {
    val temp: Array[Int] = new Array[Int](s.length)
    var i: Int = 0
    while (i < s.length) {
      temp(i) = s(i).toInt
      i += 1
    }
    temp
  }

  def calcTimeLeft(time: Array[Int], date: Array[Int]): Array[Int] = {
    var order: LocalDateTime = null
    val now = LocalDateTime.now(ZoneId.of("GMT"))
    var temp: LocalDateTime = null
    try {
      order = LocalDateTime.of(date(0), date(1), date(2), time(0), time(1), time(2))
    } catch {
      case e: Exception =>
        e.printStackTrace
        return Array[Int](0, 0, 0)
    }
    var minutes: Double = 0
    temp = order
    while (!(temp.getYear == now.getYear && temp.getDayOfMonth == now.getDayOfMonth && temp.getMonthValue == now.getMonthValue && temp.getHour == now.getHour && temp.getMinute == now.getMinute && temp.getSecond == now.getSecond)) {
      temp = temp.plusSeconds(1)
      minutes += 1
    }
    var minute = 0
    var hour = 0
    var second = 0
    hour = minutes.toInt / 3600
    minutes = minutes % 3600
    minute = minutes.toInt / 60
    second = minutes.toInt % 60
    Array[Int](hour, minute, second)
  }
}